aderyn_core/src/detect/high/high_level_calls_in_loop.rs

```diff
@@ -0,0 +1,195 @@
+use crate::context::investigator::{
+    StandardInvestigationStyle, StandardInvestigator, StandardInvestigatorVisitor,
+};
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct HighLevelCallsInLoopDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+/**
+    This [`HighLevelCallsInLoopDetector`] detector will catch address.transfer as well as high level calls in loops
+    (as they may potentially revert) and we don't specify "revert" explicitly.
+    1. address.transfer
+    2. Contract.xyz()
+
+    Things what we don't capture - `address.send`, `address.call`, etc
+    (Because we know for sure they don't revert unless we ask them to based on the return value of bool success)
+
+    The solution is to perform this function call with low level
+    call and then you can just break off the loop if the bool success returns false!
+
+    In order to catch `requires` and `reverts` in the loops, there is another detector we have in `reverts_and_requires_in_loop.rs`
+*/
+impl IssueDetector for HighLevelCallsInLoopDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for for_statement in context.for_statements() {
+            if has_high_level_calls_or_tranfers_eth_to_an_address(context, &(for_statement.into()))
+                .is_some_and(identity)
+            {
+                capture!(self, context, for_statement);
+            }
+        }
+
+        for while_statement in context.while_statements() {
+            if has_high_level_calls_or_tranfers_eth_to_an_address(
+                context,
+                &(while_statement.into()),
+            )
+            .is_some_and(identity)
+            {
+                capture!(self, context, while_statement);
+            }
+        }
+
+        for do_while_statement in context.do_while_statements() {
+            if has_high_level_calls_or_tranfers_eth_to_an_address(
+                context,
+                &(do_while_statement.into()),
+            )
+            .is_some_and(identity)
+            {
+                capture!(self, context, do_while_statement);
+            }
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::High
+    }
+
+    fn title(&self) -> String {
+        String::from("Loop contains high level calls to other contracts and/or eth is transferred.")
+    }
+
+    fn description(&self) -> String {
+        String::from("One of the dangers of having high level calls or an `address.transfer()` inside a loop is that \
```

Comment:
This sounds very different to issue ADE-54 in linear. The recommendation there is to use pull over push because it prevents DDOS attacks. Here we're saying the issue is something else. Can you explain this some more please?

Reply:
Changed to match with slither. 
✅